Command plugin can't run from CLI

### Description

As some bullet points:
* SSWG VSCode extension couldn't show OpenAPI generator's command plugin.
* Then i noticed `swift package plugin --list` doesn't include the plugin either:
```bash
% swift package plugin --list                    
‘archive’ (plugin ‘AWSLambdaPackager’ in package ‘swift-aws-lambda-runtime’)
‘format-source-code’ (plugin ‘Format Source Code’ in package ‘swift-format’)
‘generate-manual’ (plugin ‘GenerateManual’ in package ‘swift-argument-parser’)
‘lint-source-code’ (plugin ‘Lint Source Code’ in package ‘swift-format’)
```
* Running the command plugin fails anyway:
```bash
% swift package plugin generate-code-from-openapi --allow-writing-to-package-directory
/Users/mahdibm/Github/penny-bot/.build/checkouts/swift-argument-parser/Sources/ArgumentParser/Parsable Properties/Option.swift:570:94: error: multi-line string literal closing delimiter must begin on a new line
        message: """Optional @Options with default values should be declared as non-Optional."""
<Skipping a lot of similar errors>
error: fatalError
[49/153] Compiling Yams Constructor.swift
error: fatalError
```
* Running from Xcode worked perfectly as usual:
```
...
Considering target 'GitHubAPI':
- Trying OpenAPI code generation.
Swift OpenAPI Generator is running with the following configuration:
- OpenAPI document path: /Users/mahdibm/Github/penny-bot/Lambdas/GitHubAPI/openapi.yaml
- Configuration path: /Users/mahdibm/Github/penny-bot/Lambdas/GitHubAPI/openapi-generator-config.yml
- Generator modes: types, client
- Feature flags: <none>
- Output file names: Types.swift, Client.swift
- Output directory: /Users/mahdibm/Github/penny-bot/Lambdas/GitHubAPI/GeneratedSources
- Diagnostics output path: <none - logs to stderr>
- Current directory: /Users/mahdibm/Github/penny-bot
- Plugin source: command
- Is dry run: false
- Additional imports: <none>
Writing data to file Types.swift...
Writing data to file Client.swift...
- ✅ OpenAPI code generation for target 'GitHubAPI' successfully completed.
...
```

### Expected behavior

The list command should show the command plugin and the command plugin should run.

### Actual behavior

As described in Description.

### Steps to reproduce

Use https://github.com/vapor/penny-bot/releases/tag/spm-issue-6804.
Do as described in Description.

### Swift Package Manager version/commit hash

Swift Package Manager - Swift 5.9.0

### Swift & OS version (output of `swift --version ; uname -a`)

swift-driver version: 1.87.1 Apple Swift version 5.9 (swiftlang-5.9.0.128.2 clang-1500.0.40.1)
Target: arm64-apple-macosx14.0
Darwin Macbook.local 23.0.0 Darwin Kernel Version 23.0.0: Tue Aug  1 03:25:51 PDT 2023; root:xnu-10002.0.242.0.6~31/RELEASE_ARM64_T6000 arm64

--- a/Sources/Commands/PackageTools/PluginCommand.swift
+++ b/Sources/Commands/PackageTools/PluginCommand.swift
@@ -82,7 +82,7 @@ struct PluginCommand: SwiftCommand {
             let packageGraph = try swiftTool.loadPackageGraph()
             let allPlugins = PluginCommand.availableCommandPlugins(in: packageGraph)
             for plugin in allPlugins.sorted(by: { $0.name < $1.name }) {
-                guard case .command(let intent, _) = plugin.capability else { return }
+                guard case .command(let intent, _) = plugin.capability else { continue }
                 var line = "‘\(intent.invocationVerb)’ (plugin ‘\(plugin.name)’"
                 if let package = packageGraph.packages
                     .first(where: { $0.targets.contains(where: { $0.name == plugin.name }) })

